pkg/endpoint/endpoint.go

```diff
@@ -2295,11 +2291,13 @@ func (e *Endpoint) UpdateLabels(owner Owner, identityLabels, infoLabels pkgLabel
 		logfields.InfoLabels:     infoLabels.String(),
 	}).Debug("Refreshing labels of endpoint")
 
-	e.Mutex.Lock()
+	// NOTE: not sure if this should be allowed, please take a look in review
+	e.UnconditionalLock()
```

Comment:
This should check for liveness

Reply:
Should all UpdateLabels calls check for lock error? Should docker and cri `handleCreateWorkload` fail when this lock fails?

Reply:
This lock should fail as it will prevent changing the labels and triggering an identity resolution for an endpoint that is already being deleted.

Reply:
Let's review and fid the container runtime handlers separately outside of the scope of this PR. At least CRI is currently missing to hold ep.Mutex while calling `endpointmanager.UpdateReferences()` so it might potentially create a entry for an endpoint that is being deleted. Let's just make sure that we don't update the labels of the endpoint for now.